### ISSUE TITLE
[Bug] Parser hangs in infinite loop when variable delimiters are/seem not closed

### ISSUE THREAD

**Issue by mlange-42** [OinkIguana/outline#1]
Using the default delimiter settings, `outline` got stuck in an infinite loop with
```
==> Ref @{0.5}.
```

The parser interprets the dot as the end of the macro call and hangs instead of reporting an error.

I fixed it for Markdown by changing `parser/mod.rs:59ff` (added `else`):
```rust
if let Some(end_index) = input[start_index + start.len()..].find(end) {
    name.push_str(&input[..start_index]);
    name.push_str(&start);
    name.push_str(&end);
    vars.push(
        &input[start_index + start.len()..start_index + start.len() + end_index],
    );
    input = &input[start_index + start.len() + end_index + end.len()..];
} else {
    return Err(ParseError::UnclosedVariableError(orig));
}
```

However, I made other changes, rustfmt reformatted your code, and I feel too inexperienced to make a pull request.

In addition: are you interested in further developing this inspiring piece of work?
I would like to add features like file transclusion, hiding code blocks, excluding blocks from code output, and potentially per-file control over the code destination. So, I ask myself if I should work on a fork, or decouple and just derive from your work.

Currently, I test your tool in combination with the so-called [ODD protocol](https://www.ufz.de/index.php?de=40429) for describing individual-based models: https://github.com/mlange-42/outline/tree/dev/examples/netlogo_odd

Greetings!

**Comment by foxfriends** [comment]
Hey sorry for this incredibly delayed response, turns out I am not watching a lot of my own repos. I see you made a few other issues on other things, going to be fixing my github settings going forward.

If it's not too late, you are welcome to fork this and send any PRs of features you have added. I haven't really updated this thing in a long time but if you have any issues I will take a look for sure.

**Comment by mlange-42** [comment]
Hi @foxfriends,
after working on a fork for a while, I decided to "unfork" due to increasing deviation from 'outline': https://github.com/mlange-42/yarner

So far, I primarily added features for handling multiple files (transclusions, automatic inclusion of linked files, copying of source files, etc.). Further, I tried to make CLI use even simpler by adding arguments to the config file, and by a `create` command to setup a new project with some example content. I removed some things like the bird parser.

For the future, I may try to build a "reverse mode" which allows to "play back" changes of the generated code into the source documents, so people could make use of IDEs for coding.

I really appreciate the foundation and concept I found with 'outline'!
